### PR TITLE
Use separate copies of dependencies in source generators

### PR DESCRIPTION
**Pull request by captainsafia**
Addresses https://github.com/dotnet/aspnetcore/issues/32218.

### CODE REVIEW

File: src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj [hunk 1]
```diff
@@ -30,6 +31,8 @@
 
   <Target Name="GetDependencyTargetPaths">
     <ItemGroup>
+      <TargetPathWithTargetPlatformMoniker Include="$(PkgMicrosoft_AspNetCore_Razor_Language)\lib\netstandard2.0\Microsoft.AspNetCore.Razor.Language.dll" IncludeRuntimeDependency="false" />
```
Comment (pranavkm) [inline src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj:34]: To confirm, we don't have compiler binaries in this SourceGenerators dir, yes?

**Review by pranavkm — APPROVED** [review]

File: src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj (hunk 1 above)
Comment (captainsafia) [reply src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj:34]: Now it does. 😄

Comment (pranavkm) [reply src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj:34]: Sorry, I meant Microsoft.CodeAnalysis.CSharp etc

Comment (captainsafia) [reply src/RazorSdk/SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.csproj:34]: Yep -- I caught your drift. The current contents of the source-generators directory is above which is just the stuff we include in `<Analyzers>` and the EnC config file.

**Pull request merged**